### DISCUSSION TITLE
关于改进Maybe<>的Ideas

### DISCUSSION THREAD

**Discussion by SanftMonster** [Oneflow-Inc/oneflow#6830] [Ideas]
在[知乎上](https://zhuanlan.zhihu.com/p/423811115)看到oneflow要改进一下Maybe，请问目前贵团队有没有什么计划或者方向呀？

比如文中提到的```JUST```在MSVC下退化成用异常实现，目前计划首先要做的东西是这个吗？

**Comment by PragmaTwice** [comment]
感觉目前支持 msvc 的计划还不是很急切，因为除了 Maybe 还有很多其他地方不支持 msvc（比如 cmake、依赖库、一些 linux 特定的实现）。如果感兴趣可以提 PR，不过我感觉目前的构建过程可能在 windows 下跑不通。
最近的计划是把 Maybe 迁移成一个独立的项目，并且借这个机会重写 Maybe 相关的组件，移除对 shared_ptr 的依赖；可以在[这个 PR](https://github.com/Oneflow-Inc/oneflow/pull/6820)里看到目前的进展。
感谢对项目的关注，欢迎提出自己的想法和建议~
